src/paradigma/heart_rate/tfd.py

```diff
@@ -0,0 +1,553 @@
+          Dictionary containing two nested dictionaries, one for the Doppler window and one for the lag window:
+            'doppler': {
+                'win_length': Length of the Doppler-domain window.
+                'win_type': Type of Doppler-domain window function.
+                'win_param' (optional): Additional parameters for the Doppler window.
+                'win_param2' (optional): 0 for time-domain window or 1 for Doppler-domain window.
+            }
+            'lag': {
+                'win_length': Length of the lag-domain window.
+                'win_type': Type of lag-domain window function.
+                'win_param' (optional): Additional parameters for the lag window.
+                'win_param2' (optional): 0 for time-domain window or 1 for Doppler-domain window.
+            }
+          Example:
+          ```python
+          kern_params = {
+              'doppler': {
+                  'win_length': doppler_samples,
+                  'win_type': win_type_doppler,
+              },
+              'lag': {
+                  'win_length': lag_samples,
+                  'win_type': win_type_lag,
+              }
+          }
+          ```
+    Returns:
+    --------
+    tfd : ndarray
+        The computed time-frequency distribution.
+    """
+    z = get_analytic_signal(x)
+    N = len(z) // 2  # Since z is a signal of length 2N
+    Nh = int(np.ceil(N / 2))
+
+    # Generate the time-lag distribution of the analytic signal
+    tfd = gen_time_lag(z)
+
+    # Multiply the TFD by the Doppler-lag kernel
+    tfd = multiply_kernel_signal(tfd, kern_type, kern_params, N, Nh)
+    
+    # Finalize the TFD computation
+    tfd = compute_tfd(N, Nh, tfd)
+    
+    return tfd
+
+def get_analytic_signal(x):
+    """
+    Generates the signals analytic version.
+
+    Parameters:
+    -----------
+    z : ndarray
+        Input real-valued signal.
+
+    Returns:
+    --------
+    z : ndarray
+        Analytic signal with zero-padded imaginary part.
+    """
+    N = len(x)
+
+    # Ensure the signal length is even by trimming one sample if odd, since the gen_time_lag function requires an even-length signal
+    if N % 2 != 0:
+        x = x[:-1]
+
+    # Make the analytical signal of the real-valued signal z (preprocessed PPG signal)
+    # doesn't work for input of complex numbers
+    z = gen_analytic(x)  
+    
+    return z
+
+def gen_analytic(x):
+    """
+    Generates an analytic signal by zero-padding and performing FFT.
+
+    Parameters:
+    -----------
+    x : ndarray
+        Input real-valued signal.
+
+    Returns:
+    --------
+    z : ndarray
+        Analytic signal in the time domain with zeroed second half.
+    """
+    N = len(x)
+    
+    # Zero-pad the signal to double its length
+    x = np.concatenate((np.real(x), np.zeros(N)))
+    x_fft = np.fft.fft(x)
+
+    # Generate the analytic signal in the frequency domain
+    H = [1] + list(np.repeat(2, N-1)) + [1] + list(np.repeat(0, N-1))
```

Comment:
Perhaps you could use `np.concatenate` here, which works for more than 2 items as well: 

`# Define three arrays
arr1 = np.array([1, 2, 3])
arr2 = np.array([4, 5, 6])
arr3 = np.array([7, 8, 9])

# Concatenate along the first axis (default)
result = np.concatenate((arr1, arr2, arr3))
print(result)  # Output: [1 2 3 4 5 6 7 8 9]`

Or for lists:

`# Define three lists
list1 = [1, 2, 3]
list2 = [4, 5, 6]
list3 = [7, 8, 9]

# Concatenate lists with NumPy
result = np.concatenate((list1, list2, list3))
print(result)  # Output: [1 2 3 4 5 6 7 8 9]`

I suppose that the FFT works faster with two numpy arrays, but not sure.